app/components/repository-sidebar.js

```diff
@@ -68,20 +70,18 @@ export default Ember.Component.extend({
     return runningAmount + queuedAmount;
   },
 
-  @computed('features.proVersion', 'jobState.runningJobs.[]')
-  runningJobs(proVersion, runningJobs) {
-    if (!proVersion) { return []; }
+  @computed('jobState.runningJobs.[]')
+  runningJobs(runningJobs) {
     return runningJobs;
   },
 
-  @computed('features.proVersion')
-  queuedJobs(proVersion) {
-    if (!proVersion) { return []; }
-
+  @computed('currentUser')
+  queuedJobs(user) {
     const queuedStates = ['created'];
     const result = this.get('store').filter(
       'job',
-      job => queuedStates.includes(job.get('state'))
+      job =>
+        queuedStates.includes(job.get('state') && user && user.hasAccessToRepo(job.get('repo')))
```

Comment:
Hey, @drogus, since you mentioned `store.filter` [here](https://github.com/travis-ci/travis-web/pull/1205#issue-247375781), do you think it would be worth waiting for your changes there so I can make this actually work? 

The current problem is that `job.get('repo')` is asynchronous, of course, so `hasAccessToRepo` will always be false. I need to do some promise-resolving before I can add another filter, but I’m not really sure how to proceed… any ideas?

Reply:
It will work (or at least it should!), because the property [depends on permissions array](https://github.com/travis-ci/travis-web/blob/master/app/models/repo.js#L39) and `store.filter` will refresh the collection once any of the dependencies change. But please let me know if it doesn't work that way, I'm sure that it's something we can fix (because, in theory it should work).

Reply:
hmm it seems you’re right! I don’t know why it wasn’t working yesterday 😳 

![image](https://user-images.githubusercontent.com/43280/29726591-ebae3fbe-8995-11e7-9c0c-32703dbe8462.png)

Thanks for this, I should be able to make progress on it this week.